config/files/usr/share/ublue-os/just/60-custom.just

```diff
@@ -2,14 +2,85 @@
+    echo "Hardening kernel arguments applied."
 
 harden-flatpak:
-    flatpak override --user --filesystem=host-os:ro --env=LD_PRELOAD=/var/run/host/usr/lib64/libhardened_malloc.so
+    flatpak override --user --filesystem=host-os:ro --env=LD_PRELOAD=/var/run/host/usr/lib64/libhardened_malloc.so &&\
+    echo "Flatpaks are now using harened_halloc."
 
 enable-cups:
-    firewall-cmd --add-service=cups --permanent 
-    firewall-cmd --reload 
-    systemctl unmask cups
-    systemctl enable cups
-    systemctl start cups
+    firewall-cmd --add-service=cups --permanent &&\
+    firewall-cmd --reload &&\
+    systemctl unmask cups &&\
+    systemctl enable --now cups &&\
+    echo "CUPS enabled."
+
+disable-cups:
+    firewall-cmd --remove-service=cups --permanent &&\
+    firewall-cmd --reload &&\
+    systemctl stop cups &&\
+    systemctl disable cups &&\
+    systemctl mask cups &&\
+    echo "CUPS disabled."
+
+userns-on
+    # test if userns are disabled
+    if grep -q "kernel.unprivileged_userns_clone = 0" "/usr/etc/sysctl.d/hardening.conf"; then
+
+        # read the container image reference and change the string
+        userns_version=$(rpm-ostree status --json | jq -r '.deployments[0]."container-image-reference"' | sed 's/hardened/userns-&/')
+        echo "Rebasing to the user namespaces image..."
+        rpm-ostree rebase $userns_version
+        echo "Reboot for the changes to apply."
+
+    # test if userns are enabled
+    elif grep -q "kernel.unprivileged_userns_clone = 1" "/usr/etc/sysctl.d/hardening.conf"; then
+        echo "User namespaces are already enabled."
+
+    else
+        echo "The check has failed, please report that error."
+        rpm-ostree status --json | jq -r '.deployments[0]."container-image-reference"'
+        xdg-open "https://github.com/secureblue/secureblue/issues/new"
+    fi
+
+userns-off
+    # test it userns are disabled
+    if grep -q "kernel.unprivileged_userns_clone = 0" "/usr/etc/sysctl.d/hardening.conf"; then
+        echo "User namespaces are already disabled."
+
+    # test if userns are enabled
+    elif grep -q "kernel.unprivileged_userns_clone = 1" "/usr/etc/sysctl.d/hardening.conf"; then
+        nouserns_version=$(rpm-ostree status --json | jq -r '.deployments[0]."container-image-reference"' | sed 's/-userns//')
+        echo "Rebasing to the image without user namespaces..."
+        rpm-ostree rebase $nouserns_version
+        echo "Reboot for the changes to apply."
+
+    else
+        echo "The check has failed, please report that error."
+        rpm-ostree status --json | jq -r '.deployments[0]."container-image-reference"'
+        xdg-open "https://github.com/secureblue/secureblue/issues/new"
+    fi
+
+bluetooth-on
```

Comment:
this one is a good idea but instead please disable bluetooth in the image for all images, and then add the just command "enable-bluetooth"

Reply:
same thing as with cups, users have to easily be able to disable it again when not using it.

Reply:
They already can with systemctl.